src/Dapper.AOT.Analyzers/SqlAnalysis/TSqlProcessor.cs

```diff
@@ -623,6 +626,27 @@ public override void Visit(BooleanComparisonExpression node)
             base.Visit(node);
         }
 
+        public override void Visit(StringLiteral node)
+        {
+            foreach (var token in node.ScriptTokenStream.Where(t => t.TokenType == TSqlTokenType.AsciiStringLiteral))
+            {
+                // searching for 2 appearances of '?'
+                int firstAppearance;
+                if ((firstAppearance = token.Text.IndexOf('?')) >= 0)
```

Comment:
just to prevent too much indenting, maybe we can `&&` some of this?

``` c#
if ((firstAppearance = token.Text.IndexOf('?')) >= 0
    && token.Text.IndexOf('?', firstAppearance + 1) >= 0
    && CompiledRegex.PseudoPositional.IsMatch(node.Value))
```